### ISSUE TITLE
REDUCE, syntax false positive

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1997]
`lv_idx not found, findTop(check_syntax)`

```abap
TYPES ty_t_coefficients TYPE STANDARD TABLE OF i WITH DEFAULT KEY.
      
DATA it_coefficients TYPE ty_t_coefficients.

DATA(rv_val) = REDUCE i( INIT n = 0
                       FOR coef IN it_coefficients
                       INDEX INTO lv_idx
                       LET lv_pos = lv_idx - 1
                       IN NEXT n = n + coef * mv_number+lv_pos(1) ).
```

**Comment by victorizbitskiy** [comment]
Nice and fast work, @larshp! 👍 
![giphy](https://user-images.githubusercontent.com/62376906/120226325-a4eafa80-c24f-11eb-9131-26101584ca59.gif)

**Comment by larshp** [comment]
there is always another false positive, but fixing it one by one 🤠
